unit-tests/src/test/scala/java/lang/DoubleSuite.scala

```diff
@@ -195,10 +195,58 @@ object DoubleSuite extends tests.Suite {
     assert(Double.parseDouble("Infinity") == Double.POSITIVE_INFINITY)
     assert(Double.parseDouble("-Infinity") == Double.NEGATIVE_INFINITY)
     assert(Double.isNaN(Double.parseDouble("NaN")))
+
+    val epsilon = 0.0001
+    assert(Math.abs(Double.parseDouble("6.66D") - 6.66) < epsilon, "a8")
+
+    // Java allows trailing whitespace, including tabs.
+    assert(Math.abs(Double.parseDouble("6.66D\t ") - 6.66) < epsilon, "a9")
+
+    assert(Math.abs(Double.parseDouble("6.66d") - 6.66) < epsilon, "a10")
+
+    assert(Math.abs(Double.parseDouble("7.77F") - 7.77) < epsilon, "a11")
+    assert(Math.abs(Double.parseDouble("7.77f") - 7.77) < epsilon, "a12")
```

Comment:
`parseDouble` has a very precise definition of its result. We should test it with a strong `==`, not with an `epsilon` difference.

Reply:
Please add tests with a `\u0000` character, such as `"7.77\u0000 "` (should succeed) and `"7.77\u0000a"` (should fail).

Reply:
In both DoubleSuite.scala & FloatSuite.scala:
* `epsilon` comparisons changed to strong `==` test as recommended.
* Requested tests added to each file.

I also fixed a copy & paste error in FloatSuite.scala where Double.parseDouble() was being called instead of 
the correct Float.parseFloat().